modules/apps/27-interchain-accounts/types/packet.go

```diff
@@ -57,6 +57,10 @@ func (iapd InterchainAccountPacketData) GetBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&iapd))
 }
 
+func (iapd *InterchainAccountPacketData) UnmarshalJSON(bz []byte) error {
```

Comment:
I am fine with this approach, but maybe an alternative could be to have instead and constructor function if that's possible? Like this:

```go
func NewInterchainAccountPacketData(bz []byte) (InterchainAccountPacketData, error) {
  var packetData icatypes.InterchainAccountPacketData
  if err := ModuleCdc.UnmarshalJSON(bz, &packetData); err != nil {
    return nil, err
  }
  return packetData, nil
}
```

It just feels more natural to me that declaring the value and then mutating it. That way we don't mix value and pointer receiver functions in the type, which I think it's another go proverb? :)

Reply:
This is exactly what I did originally, but @damiannolan and @DimitrisJim were slightly favouring the current approach. I'm curious what others think. Regardless, I apparently can't merge this until ica code owners approve it

Reply:
Ah, funny, haha! :) Well, if the majority prefers the current approach, I am ok with it.